### ISSUE TITLE
loss nan for pisa

### ISSUE THREAD

**Issue by qih96** [open-mmlab/mmdetection#3062]
2020-06-17 13:16:06,637 - mmdet - INFO - Environment info:
------------------------------------------------------------
sys.platform: linux
Python: 3.7.0 (default, Oct  9 2018, 10:31:47) [GCC 7.3.0]
CUDA available: True
CUDA_HOME: /usr/local/cuda
NVCC: Cuda compilation tools, release 10.1, V10.1.105
GPU 0,1,2,3: TITAN Xp
GCC: gcc (Ubuntu 5.4.0-6ubuntu1~16.04.12) 5.4.0 20160609
PyTorch: 1.5.0
PyTorch compiling details: PyTorch built with:
  - GCC 7.3
  - C++ Version: 201402
  - Intel(R) Math Kernel Library Version 2020.0.1 Product Build 20200208 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v0.21.1 (Git Hash 7d2fd500bc78936d1d648ca713b901012f470dbc)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - CUDA Runtime 10.1
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_61,code=sm_61;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_37,code=compute_37
  - CuDNN 7.6.3
  - Magma 2.5.2
  - Build settings: BLAS=MKL, BUILD_TYPE=Release, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -fopenmp -DNDEBUG -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DUSE_INTERNAL_THREADPOOL_IMPL -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, USE_CUDA=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, USE_STATIC_DISPATCH=OFF, 

TorchVision: 0.6.0a0+82fd1c8
OpenCV: 4.2.0
MMCV: 0.5.9
MMDetection: 2.1.0+1ae61f9
MMDetection Compiler: GCC 5.4
MMDetection CUDA Compiler: 10.1
------------------------------------------------------------

2020-06-17 13:16:06,637 - mmdet - INFO - Distributed training: True
2020-06-17 13:16:06,833 - mmdet - INFO - Config:
model = dict(
    type='RetinaNet',
    pretrained='open-mmlab://resnext101_32x4d',
    backbone=dict(
        type='ResNeXt',
        depth=101,
        num_stages=4,
        out_indices=(0, 1, 2, 3),
        frozen_stages=1,
        norm_cfg=dict(type='BN', requires_grad=True),
        norm_eval=True,
        style='pytorch',
        groups=32,
        base_width=4),
    neck=dict(
        type='FPN',
        in_channels=[256, 512, 1024, 2048],
        out_channels=256,
        start_level=1,
        add_extra_convs='on_input',
        num_outs=5),
    bbox_head=dict(
        type='PISARetinaHead',
        num_classes=80,
        in_channels=256,
        stacked_convs=4,
        feat_channels=256,
        anchor_generator=dict(
            type='AnchorGenerator',
            octave_base_scale=4,
            scales_per_octave=3,
            ratios=[0.5, 1.0, 2.0],
            strides=[8, 16, 32, 64, 128]),
        bbox_coder=dict(
            type='DeltaXYWHBBoxCoder',
            target_means=[0.0, 0.0, 0.0, 0.0],
            target_stds=[1.0, 1.0, 1.0, 1.0]),
        loss_cls=dict(
            type='FocalLoss',
            use_sigmoid=True,
            gamma=2.0,
            alpha=0.25,
            loss_weight=1.0),
        loss_bbox=dict(type='SmoothL1Loss', loss_weight=1.0, beta=0.11)))
train_cfg = dict(
    assigner=dict(
        type='MaxIoUAssigner',
        pos_iou_thr=0.5,
        neg_iou_thr=0.4,
        min_pos_iou=0,
        ignore_iof_thr=-1),
    allowed_border=-1,
    pos_weight=-1,
    debug=False,
    isr=dict(k=2.0, bias=0.0),
    carl=dict(k=1.0, bias=0.2))
test_cfg = dict(
    nms_pre=1000,
    min_bbox_size=0,
    score_thr=0.05,
    nms=dict(type='nms', iou_thr=0.5),
    max_per_img=100)
dataset_type = 'CocoDataset'
data_root = '../coco/'
img_norm_cfg = dict(
    mean=[123.675, 116.28, 103.53], std=[58.395, 57.12, 57.375], to_rgb=True)
train_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(type='LoadAnnotations', with_bbox=True),
    dict(type='Resize', img_scale=(1333, 800), keep_ratio=True),
    dict(type='RandomFlip', flip_ratio=0.5),
    dict(
        type='Normalize',
        mean=[123.675, 116.28, 103.53],
        std=[58.395, 57.12, 57.375],
        to_rgb=True),
    dict(type='Pad', size_divisor=32),
    dict(type='DefaultFormatBundle'),
    dict(type='Collect', keys=['img', 'gt_bboxes', 'gt_labels'])
]
test_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(
        type='MultiScaleFlipAug',
        img_scale=(1333, 800),
        flip=False,
        transforms=[
            dict(type='Resize', keep_ratio=True),
            dict(type='RandomFlip'),
            dict(
                type='Normalize',
                mean=[123.675, 116.28, 103.53],
                std=[58.395, 57.12, 57.375],
                to_rgb=True),
            dict(type='Pad', size_divisor=32),
            dict(type='ImageToTensor', keys=['img']),
            dict(type='Collect', keys=['img'])
        ])
]
data = dict(
    samples_per_gpu=2,
    workers_per_gpu=2,
    train=dict(
        type='CocoDataset',
        ann_file='../coco/annotations/instances_train2017.json',
        img_prefix='../coco/train2017/',
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(type='LoadAnnotations', with_bbox=True),
            dict(type='Resize', img_scale=(1333, 800), keep_ratio=True),
            dict(type='RandomFlip', flip_ratio=0.5),
            dict(
                type='Normalize',
                mean=[123.675, 116.28, 103.53],
                std=[58.395, 57.12, 57.375],
                to_rgb=True),
            dict(type='Pad', size_divisor=32),
            dict(type='DefaultFormatBundle'),
            dict(type='Collect', keys=['img', 'gt_bboxes', 'gt_labels'])
        ]),
    val=dict(
        type='CocoDataset',
        ann_file='../coco/annotations/instances_val2017.json',
        img_prefix='../coco/val2017/',
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(
                type='MultiScaleFlipAug',
                img_scale=(1333, 800),
                flip=False,
                transforms=[
                    dict(type='Resize', keep_ratio=True),
                    dict(type='RandomFlip'),
                    dict(
                        type='Normalize',
                        mean=[123.675, 116.28, 103.53],
                        std=[58.395, 57.12, 57.375],
                        to_rgb=True),
                    dict(type='Pad', size_divisor=32),
                    dict(type='ImageToTensor', keys=['img']),
                    dict(type='Collect', keys=['img'])
                ])
        ]),
    test=dict(
        type='CocoDataset',
        ann_file='../coco/annotations/instances_val2017.json',
        img_prefix='../coco/val2017/',
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(
                type='MultiScaleFlipAug',
                img_scale=(1333, 800),
                flip=False,
                transforms=[
                    dict(type='Resize', keep_ratio=True),
                    dict(type='RandomFlip'),
                    dict(
                        type='Normalize',
                        mean=[123.675, 116.28, 103.53],
                        std=[58.395, 57.12, 57.375],
                        to_rgb=True),
                    dict(type='Pad', size_divisor=32),
                    dict(type='ImageToTensor', keys=['img']),
                    dict(type='Collect', keys=['img'])
                ])
        ]))
evaluation = dict(interval=1, metric='bbox')
optimizer = dict(type='SGD', lr=0.01, momentum=0.9, weight_decay=0.0001)
optimizer_config = dict(grad_clip=None)
lr_config = dict(
    policy='step',
    warmup='linear',
    warmup_iters=500,
    warmup_ratio=0.001,
    step=[8, 11])
total_epochs = 12
checkpoint_config = dict(interval=1)
log_config = dict(interval=50, hooks=[dict(type='TextLoggerHook')])
dist_params = dict(backend='nccl')
log_level = 'INFO'
load_from = None
resume_from = None
workflow = [('train', 1)]
work_dir = './work_dirs/pisa_retinanet_x101_32x4d_fpn_1x_coco'
gpu_ids = range(0, 1)

2020-06-17 13:16:07,419 - mmdet - INFO - load model from: open-mmlab://resnext101_32x4d
2020-06-17 13:16:24,276 - mmdet - INFO - workflow: [('train', 1)], max: 12 epochs
2020-06-17 13:17:30,663 - mmdet - INFO - Epoch [1][50/14659]	lr: 0.00099, eta: 2 days, 16:50:58, time: 1.328, data_time: 0.585, memory: 6774, loss_cls: 1.2292, loss_bbox: 0.5160, loss_carl: 0.5160, loss: 2.2613
2020-06-17 13:18:08,700 - mmdet - INFO - Epoch [1][100/14659]	lr: 0.00199, eta: 2 days, 2:59:27, time: 0.761, data_time: 0.005, memory: 6774, loss_cls: 1.2361, loss_bbox: 0.5178, loss_carl: 0.5178, loss: 2.2718
2020-06-17 13:18:47,807 - mmdet - INFO - Epoch [1][150/14659]	lr: 0.00299, eta: 1 day, 22:42:45, time: 0.782, data_time: 0.005, memory: 6774, loss_cls: 1.2152, loss_bbox: 0.4933, loss_carl: 0.4933, loss: 2.2017
2020-06-17 13:19:27,114 - mmdet - INFO - Epoch [1][200/14659]	lr: 0.00399, eta: 1 day, 20:37:00, time: 0.786, data_time: 0.005, memory: 6774, loss_cls: 1.2164, loss_bbox: 0.4911, loss_carl: 0.4911, loss: 2.1986
2020-06-17 13:20:06,086 - mmdet - INFO - Epoch [1][250/14659]	lr: 0.00499, eta: 1 day, 19:17:22, time: 0.779, data_time: 0.005, memory: 6774, loss_cls: 1.1016, loss_bbox: 0.4786, loss_carl: 0.4753, loss: 2.0554
2020-06-17 13:20:45,328 - mmdet - INFO - Epoch [1][300/14659]	lr: 0.00598, eta: 1 day, 18:26:42, time: 0.785, data_time: 0.005, memory: 6774, loss_cls: 1.0873, loss_bbox: 0.4549, loss_carl: 0.4492, loss: 1.9915
2020-06-17 13:21:24,719 - mmdet - INFO - Epoch [1][350/14659]	lr: 0.00698, eta: 1 day, 17:51:34, time: 0.788, data_time: 0.005, memory: 6774, loss_cls: 1.0374, loss_bbox: 0.4590, loss_carl: 0.4523, loss: 1.9486
2020-06-17 13:22:03,660 - mmdet - INFO - Epoch [1][400/14659]	lr: 0.00798, eta: 1 day, 17:21:46, time: 0.779, data_time: 0.005, memory: 6774, loss_cls: 1.1719, loss_bbox: 0.4575, loss_carl: 0.4562, loss: 2.0856
2020-06-17 13:22:42,682 - mmdet - INFO - Epoch [1][450/14659]	lr: 0.00898, eta: 1 day, 16:58:58, time: 0.780, data_time: 0.005, memory: 6774, loss_cls: 1.2173, loss_bbox: 0.4668, loss_carl: 0.4668, loss: 2.1509
2020-06-17 13:23:22,036 - mmdet - INFO - Epoch [1][500/14659]	lr: 0.00998, eta: 1 day, 16:42:32, time: 0.787, data_time: 0.005, memory: 6774, loss_cls: 1.2117, loss_bbox: 0.4940, loss_carl: 0.4938, loss: 2.1995
2020-06-17 13:24:00,836 - mmdet - INFO - Epoch [1][550/14659]	lr: 0.01000, eta: 1 day, 16:26:02, time: 0.776, data_time: 0.005, memory: 6774, loss_cls: nan, loss_bbox: nan, loss_carl: nan, loss: nan
2020-06-17 13:24:39,785 - mmdet - INFO - Epoch [1][600/14659]	lr: 0.01000, eta: 1 day, 16:12:54, time: 0.779, data_time: 0.005, memory: 6774, loss_cls: nan, loss_bbox: nan, loss_carl: nan, loss: nan
2020-06-17 13:25:18,160 - mmdet - INFO - Epoch [1][650/14659]	lr: 0.01000, eta: 1 day, 15:59:06, time: 0.767, data_time: 0.004, memory: 6774, loss_cls: nan, loss_bbox: nan, loss_carl: nan, loss: nan
2020-06-17 13:25:56,970 - mmdet - INFO - Epoch [1][700/14659]	lr: 0.01000, eta: 1 day, 15:49:00, time: 0.776, data_time: 0.005, memory: 6774, loss_cls: nan, loss_bbox: nan, loss_carl: nan, loss: nan
2020-06-17 13:26:35,915 - mmdet - INFO - Epoch [1][750/14659]	lr: 0.01000, eta: 1 day, 15:40:41, time: 0.779, data_time: 0.005, memory: 6774, loss_cls: nan, loss_bbox: nan, loss_carl: nan, loss: nan
2020-06-17 13:27:14,500 - mmdet - INFO - Epoch [1][800/14659]	lr: 0.01000, eta: 1 day, 15:32:00, time: 0.772, data_time: 0.005, memory: 6774, loss_cls: nan, loss_bbox: nan, loss_carl: nan, loss: nan
2020-06-17 13:27:53,414 - mmdet - INFO - Epoch [1][850/14659]	lr: 0.01000, eta: 1 day, 15:25:25, time: 0.778, data_time: 0.005, memory: 6774, loss_cls: nan, loss_bbox: nan, loss_carl: nan, loss: nan
2020-06-17 13:28:31,708 - mmdet - INFO - Epoch [1][900/14659]	lr: 0.01000, eta: 1 day, 15:17:28, time: 0.766, data_time: 0.005, memory: 6774, loss_cls: nan, loss_bbox: nan, loss_carl: nan, loss: nan
2020-06-17 13:29:10,806 - mmdet - INFO - Epoch [1][950/14659]	lr: 0.01000, eta: 1 day, 15:12:46, time: 0.782, data_time: 0.005, memory: 6774, loss_cls: nan, loss_bbox: nan, loss_carl: nan, loss: nan
2020-06-17 13:29:49,587 - mmdet - INFO - Epoch [1][1000/14659]	lr: 0.01000, eta: 1 day, 15:07:32, time: 0.776, data_time: 0.005, memory: 6774, loss_cls: nan, loss_bbox: nan, loss_carl: nan, loss: nan
2020-06-17 13:30:28,643 - mmdet - INFO - Epoch [1][1050/14659]	lr: 0.01000, eta: 1 day, 15:03:30, time: 0.781, data_time: 0.005, memory: 6774, loss_cls: nan, loss_bbox: nan, loss_carl: nan, loss: nan
2020-06-17 13:31:07,380 - mmdet - INFO - Epoch [1][1100/14659]	lr: 0.01000, eta: 1 day, 14:58:56, time: 0.775, data_time: 0.005, memory: 6774, loss_cls: nan, loss_bbox: nan, loss_carl: nan, loss: nan
2020-06-17 13:31:46,346 - mmdet - INFO - Epoch [1][1150/14659]	lr: 0.01000, eta: 1 day, 14:55:18, time: 0.779, data_time: 0.005, memory: 6774, loss_cls: nan, loss_bbox: nan, loss_carl: nan, loss: nan
2020-06-17 13:32:25,186 - mmdet - INFO - Epoch [1][1200/14659]	lr: 0.01000, eta: 1 day, 14:51:36, time: 0.777, data_time: 0.005, memory: 6774, loss_cls: nan, loss_bbox: nan, loss_carl: nan, loss: nan
2020-06-17 13:33:03,828 - mmdet - INFO - Epoch [1][1250/14659]	lr: 0.01000, eta: 1 day, 14:47:41, time: 0.773, data_time: 0.005, memory: 6774, loss_cls: nan, loss_bbox: nan, loss_carl: nan, loss: nan

**Comment by yhcao6** [comment]
It seems you are using 4 GPUs. Then you need to downscale learning rate by 2.

**Comment by qih96** [comment]
@yhcao6  Thanks for your reply, I will try it later

**Comment by qih96** [comment]
@yhcao6 it works for me, thank you

**Comment by Daniel123jia** [comment]
Hello, my learning rate is set to 0.001, but ‘nan’ appears at the end of the first round of fast training. What is your final learning rate?

**Comment by Daniel123jia** [comment]
> @yhcao6 it works for me, thank you

Hello, my learning rate is set to 0.001, but nan appears at the end of the first epoch. What is your final learning rate?

**Comment by qih96** [comment]
@Daniel123jia my learning rate is set to 0.005.

**Comment by Daniel123jia** [comment]
thank you !



At 2020-06-22 14:39:40, "qih96" <notifications@github.com> wrote:

@Daniel123jia my learning rate is set to 0.005.
